### PR TITLE
First fixes from yargs PR

### PR DESCRIPTION
**Pull request by rylandg**
Test via the spec tests and sdk tests. Invoked all valid commands manually without issue.

These are the first set of fixes based on these comments

https://github.com/gobinaris/binaris/pull/238

### CODE REVIEW

**Comment by arielshaqed** [comment]
Taking this as DDR, but I don't think I'm the original reviewer for most of the comments that this fixes.  @rylandg consider adding as reviewer people who know what it's all about, so we don't go around in circles.

File: cli.js [hunk 1]
```diff
@@ -1,27 +1,30 @@
...
+  process.exit(1);
+};
+
 const handleCommand = async function handleCommand(options, specificHandler) {
-  const numArgs = options._.length;
-  if (numArgs > 1) {
-    yargs.showHelp();
-    logger.error('Too many positional args given for command');
-    process.exit(1);
+  // eslint-disable-next-line no-param-reassign
```
Comment (arielshaqed) [inline cli.js]: Why is this needed, I don't see param reassignment here.

File: cli.js [hunk 2]
```diff
@@ -123,10 +119,10 @@ Usage: $0 <command> [options]` // eslint-disable-line comma-dangle
   .wrap(null);
 
 const commands = yargs.getCommandInstance().getCommands();
-const argv = yargs.argv;
+// first command pushed to command stack(from user input)
```
Comment (arielshaqed) [inline cli.js]: What strange wizardry is this?  It looks like you're using `getCommandInstance` (which I couldn't find in the API docs), in order to manually perform behaviour I would expect to get from `yargs`.

Comment (rylandg) [reply cli.js]: Yea that wizardry is basically what `commander` feels like all the time. I'm not sure why it's not directly supported.

File: cli.js (hunk 1 above)
Comment (arielshaqed) [reply cli.js]: 🏓 ping?

File: cli.js (hunk 2 above)
Comment (arielshaqed) [reply cli.js]: "Welcome to the future," basically :-(.

File: cli.js (hunk 1 above)
Comment (rylandg) [reply cli.js]: Sorry was busying focusing on other things, I'll remove it in a subsequent patch.
